src/codeu/chat/client/core/View.java

```diff
@@ -136,4 +140,26 @@ public View(ConnectionSource source) {
 
     return messages;
   }
+
+  @Override
+  public ServerInfo getInfo() {
+    try (final Connection connection = this.source.connect()) {
```

Comment:
just "source.connect()". It's usually fine to leave 'this' implied except in cases of variable shadowing, and it's consistent with the rest of this class.